modules/core/src/main/java/org/apache/synapse/mediators/builtin/LogMediator.java

```diff
@@ -244,13 +244,9 @@ public String getSeparator() {
     }
 
     public void setSeparator(String separator) {
-        if (separator.equals("\\n")) {
-            this.separator = "\n";
-        } else {
-            this.separator = separator;
-        }
+        this.separator = separator.replace("\\n", "\n").replace("\\t", "\t");
```

Comment:
Yes, lets check the possible special characters and update this. 
I guess the best option would be providing a drop down to select the log separator, rather than providing a text box to enter it. But may be this is too late for this change